### PR TITLE
Bmark_solvers with solver_specific stats bug

### PR DESCRIPTION
**Pull request by MaxenceGollier**
Discussed in #170

### CODE REVIEW

File: src/run_solver.jl [hunk 1]
```diff
@@ -161,27 +185,34 @@ function solve_problems(
...
+        end
       finally
         finalize(problem)
       end
     end
-    (skipthis && prune) || @info log_row(stats[end, col_idx])
+    ((skipthis && prune) || first_problem) || @info log_row(stats[end, col_idx])
+    if colstats == [:solver_name, :name, :nvar, :ncon, :status, :iter, :elapsed_time, :objective, :dual_feas, :primal_feas,]
+      !first_problem || @info log_row(Any[solver_name, problem.meta.name, problem.meta.nvar, problem.meta.ncon, :exception, 0, Inf, Inf, Inf, Inf])
+    end # We don't print the first problems that have exceptions/skips when colstats is not default...
```
Comment (MaxenceGollier) [inline src/run_solver.jl:215]: I don't know if my approach is optimal, what might happen here is that colstats does not have its default value, then if the n first iterations are skipped/throw an error, then the first n iterations will not be printed.

**Comment by tmigot** [comment]
Thanks @MaxenceGollier for the PR. I am just wondering if counting the number of unsuccessful before the first solve add updating this line:
https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/blob/360eb42a04beca2a57ee6e894481ebe60695a6b0/src/run_solver.jl#L129
with `k => Vector{Union{typeof(v), Missing}}(undef, nb_unsuccessful),` woudl have been a simpler solution?

**Comment by MaxenceGollier** [comment]
> Thanks @MaxenceGollier for the PR. I am just wondering if counting the number of unsuccessful before the first solve add updating this line:
> 
> https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/blob/360eb42a04beca2a57ee6e894481ebe60695a6b0/src/run_solver.jl#L129
> 
> 
> with `k => Vector{Union{typeof(v), Missing}}(undef, nb_unsuccessful),` woudl have been a simpler solution?

Definitely, thank you.

**Review by tmigot — APPROVED** [review]
LGTM, thanks @MaxenceGollier

**Comment by MaxenceGollier** [comment]
@dpo can you take a look and merge if you think this is good ?

**Comment by dpo** [comment]
What is this error? https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/actions/runs/11801745210/job/32891059863?pr=171#step:9:15

File: test/test_bmark.jl [hunk 2]
```diff
@@ -91,6 +91,59 @@ function test_bmark()
...
+    @test stats[:dummy_solver_specific][3,:status] == :exception
+    @test size(stats[:dummy_solver_specific], 1) == 3
+
+    stats = bmark_solvers(
+      solvers, 
+      problems, 
+      prune = false, 
+      skipif = problem -> begin
+      return problem.meta.ncon == 0
+      end
```
Comment (dpo) [inline test/test_bmark.jl:139]:
```suggestion
      skipif = problem -> problem.meta.ncon == 0,
```

File: test/test_bmark.jl [hunk 3]
```diff
@@ -91,6 +91,59 @@ function test_bmark()
...
+      
+    ]
+
+    solvers = Dict(
+      :dummy_solver_specific => 
+        nlp -> dummy_solver(
+          nlp,
+          callback = (nlp, solver, stats) -> begin
+            set_solver_specific!(stats, :foo, 1)
+          end
```
Comment (dpo) [inline test/test_bmark.jl:123]:
```suggestion
          callback = (nlp, solver, stats) -> set_solver_specific!(stats, :foo, 1),
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by tmigot** [comment]
> What is this error? https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/actions/runs/11801745210/job/32891059863?pr=171#step:9:15

https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/issues/172

**Comment by farhadrclass** [comment]
@MaxenceGollier  does this fix our #167 issue as well?

**Review by dpo — APPROVED** [review]

**Pull request merged**